etc/taskcluster/decisionlib.py

```diff
@@ -173,6 +173,22 @@ def with_treeherder(self, category, symbol=None):
         self.treeherder_required = False  # Taken care of
         return self
 
+    def with_s3_upload_secret(self):
+        return (
+            self
+            .with_scopes("secrets:get:project/servo/s3-upload")
+            .with_env(PY=r"""if 1:
+                import urllib, json, os
+                from os.path import expanduser, join
+                url = "http://taskcluster/secrets/v1/secret/project/servo/s3-upload"
+                secret = json.load(urllib.urlopen(url))["secret"]
```

Comment:
No need for this to block this PR, but it would be cleaner and perhaps less fragile to move this bit of Python to `python/servo/package_commands.py`, add a `--secret-from-taskcluster` flag to `./mach upload-nightly`, and [pass the keys as parameters to `boto3.client()`](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#method-parameters) instead of writing a file.

Reply:
I liked this idea so much that I implemented it.